arch/riscv/core/smp.c

```diff
@@ -86,14 +86,16 @@ static atomic_val_t cpu_pending_ipi[CONFIG_MP_MAX_NUM_CPUS];
 #define IPI_SCHED	0
 #define IPI_FPU_FLUSH	1
 
-void arch_sched_ipi(void)
+void arch_sched_ipi(uint32_t cpu_bitmap)
 {
 	unsigned int key = arch_irq_lock();
 	unsigned int id = _current_cpu->id;
 	unsigned int num_cpus = arch_num_cpus();
+	uint32_t bit = 1;
 
-	for (unsigned int i = 0; i < num_cpus; i++) {
-		if (i != id && _kernel.cpus[i].arch.online) {
+	for (unsigned int i = 0; i < num_cpus; i++, bit <<= 1) {
```

Comment:
Is uint32_t better?

Reply:
I just left the type unchanged figuring that even if it should be, it was probably outside the scope of what I wanted to change. Aside: I may be wrong but I think that all of our currently supported architectures have an equivalency between 'unsigned int' and 'uint32_t'.

Reply:
> Is uint32_t better?

If it were, then it would be what `arch_num_cpus()` returns?

Fixed-width types are great for protocols and interacting with hardware but why should they be used in small `for` loops? This could confuse the reader into thinking the width matters here.

I didn't find any https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html about this but maybe I missed one.

Reply:
FWIW for a while Zephyr was trying to get serious about MISRA compliance, which disallows variable size types like int and long and requires the stdint explicit ones.  Not sure what the policy is these days.  The existing code remains sort of a mix depending on the specific archaeology of the subsystem.

Reply:
> >  I didn't find any https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html about this but maybe I missed one.

> FWIW for a while Zephyr was trying to get serious about MISRA compliance,  ...

For anyone who didn't open it, that page is mostly MISRA policies.